hooks/pre-commit

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+
+echo "Uncrustifying staged files..."
+
+RELPATH=../..
+staged=$(git diff --name-only --cached)
+
+cd ./tools/codestyle/
+while IFS= read -r file; do
+	ext="${file##*.}"
+	if [ $ext != "cpp" ] && [ $ext != "h" ]; then
```

Comment:
Wouldn't we use the 4 space indenting here too?

Reply:
for shell scripts, I've always used tabs. I thought our coding rules apply only to c/cpp :man_shrugging:

Reply:
For now at least. Other she'll scripts don't use tabs though

Reply:
`build/run_tests.sh` uses 2 spaces. `build/Linux+BSD/mscore.1.preview.sh` uses a mix of tabs and 4 spaces. I think it's safe to say that we don't have a code style for shell scripts.

Reply:
Then it seems about time to establish one. As we use 4 spaces indent in code, we should in scripts too

Reply:
That's a PR for you to make, then :smiley: